### ISSUE TITLE
Workaround for Docker Desktop Windows Image Build Issue Post-Version 4.27.2 Update

### ISSUE THREAD

**Issue by urispmts** [docker/for-win#13949]
## Workaround for Docker Desktop Windows Image Build Issue Post-Version 4.27.2 Update

**TL;DR:** If you're experiencing build issues with Docker Desktop on Windows after the 4.27.2 update, try closing Visual Studio 2022 if it has the project open that you're attempting to build. This may resolve file locking issues that prevent the build from completing.

**Description:**  
After updating to Docker Desktop version 4.27.2 on Windows, significant challenges were encountered during the Windows image build process. Despite experimenting with both WSL2 and Hyper-V settings and adjusting available resources, the issues persisted. Following recommendations, a complete purge of Docker assets was performed, which allowed for a single successful image build before the problem resurfaced.

The unexpected breakthrough occurred upon closing Visual Studio 2022, which had the project open that was being attempted to build. Remarkably, closing Visual Studio allowed the build to complete without errors, suggesting a potential file locking issue. It appears this issue is isolated to the project currently being built; other instances of Visual Studio with different projects open do not trigger the problem. The build commands were executed through PowerShell using the Docker CLI.

**Purpose of Sharing:**  
The aim of sharing this experience is to highlight a workaround that might not be immediately obvious due to the cryptic nature of the encountered errors. If you're facing similar build issues, consider closing the instance of Visual Studio that has the project open. This approach may save considerable time and effort in troubleshooting.

### Reproduce

```pwsh
docker build --platform amd64 -t <image-name> -m 2g -f Test-Dockerfile .
```

### Expected behavior

docker build should build the image according to the Dockerfile supplied with no errors

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.10
 Version:           25.0.3
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        4debf41
 Built:             Tue Feb  6 21:13:02 2024
 OS/Arch:           windows/amd64
 Context:           default

Server: Docker Desktop 4.27.2 (137060)
 Engine:
  Version:          25.0.3
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       f417435
  Built:            Tue Feb  6 20:55:49 2024
  OS/Arch:          windows/amd64
  Experimental:     false
```

### docker info

```bash
Client:
 Version:    25.0.3
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1-desktop.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.5-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.24
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.21
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.0.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.4.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 2
  Running: 0
  Paused: 0
  Stopped: 2
 Images: 337
 Server Version: 25.0.3
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 19045 (19041.1.amd64fre.vb_release.191206-1406)
 Operating System: Microsoft Windows Version 22H2 (OS Build 19045.4046)
 OSType: windows
 Architecture: x86_64
 CPUs: 16
 Total Memory: 31.39GiB
 Name: uris11
 ID: 0c3e6e02-cdf1-4fd1-b567-ef4fb59beaca
 Docker Root Dir: C:\ProgramData\Docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

### Diagnostics ID

C1C8D5E1-CCF0-4835-ACEA-1B8D5C0ED402/20240304130418

### Additional Info

Sample error messages:
```pwsh
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\schemes\\urn.js.map to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\schemes\\ws.d.ts to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\schemes\\ws.js to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\schemes\\ws.js.map to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\schemes\\wss.d.ts to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\schemes\\wss.js to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\schemes\\wss.js.map to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\uri.d.ts to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\uri.js to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\uri.js.map to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\util.d.ts to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\util.js to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\dist\\esnext\\util.js.map to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\package.json to tar: archive/tar: missed writing 4239 bytes"
time="2024-03-04T14:32:56+00:00" level=error msg="Can't add file \\\\?\\C:\\_git\\prj1\\web\\ui\\node_modules\\uri-js\\yarn.lock to tar: archive/tar: missed writing 4239 bytes"
```

**Comment by urispmts** [comment]
The issue from 4.27.2 remains in 4.28.0; the 4.27.2 workaround still works.

**Comment by catmanjan** [comment]
How can this still be an issue...
